packages/fast-colors/src/component-state-color-palette.ts

```diff
@@ -0,0 +1,165 @@
+import { ColorRGBA64 } from "./color-rgba-64";
+import { ColorPalette } from "./color-palette";
+import { hslToRGB, rgbToHSL, rgbToLuminanceLinear } from "./color-converters";
+import { ColorScale, ColorScaleStop } from "./color-scale";
+import { ColorHSL } from "./color-hsl";
+import { ColorInterpolationSpace } from "./color-interpolation";
+import { parseColorHexRGB } from "./parse-color";
+
+export interface ComponentStateColorPaletteConfig {
+    baseColor?: ColorRGBA64;
+    steps?: number;
+}
+
+export class ComponentStateColorPalette {
+    public static readonly defaultPaletteConfig: ComponentStateColorPaletteConfig = {
+        baseColor: parseColorHexRGB("#808080"),
+        steps: 94,
+    };
+
+    public palette: ColorRGBA64[] = [];
+    private readonly config: ComponentStateColorPaletteConfig;
+
+    constructor(config: ComponentStateColorPaletteConfig) {
```

Comment:
because both of these arguments are optional, I'd recommend making the argument also optional or assigning a default. As is, I have to do:

```new ComponentStateColorPalette({})```
to satisfy typescript, whereas I'd expect to be able to do
```new ComponentStateColorPalette()```
since none of the config data is actually required.